### PR TITLE
Use a different status code for "No notifiers"

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix owncloud/core#19502

@davitol

### CODE REVIEW

**Comment by karlitschek** [comment]
makes sense :+1:

**Comment by nickvergessen** [comment]
Added to the [client docs page](https://github.com/owncloud/notifications/wiki/Reading-and-deleting-notifications-as-a-Client):
> **Note:** In case the Request return status code 204 (No content), you can slow down the polling to once per hour. This status code means that there is no app, that can generate notifications.

File: controller/endpointcontroller.php [hunk 1]
```diff
@@ -70,7 +70,7 @@ public function get() {
 		// We stop polling for them.
 		if (!$this->manager->hasNotifiers()) {
 			$response = new Response();
-			$response->setStatus(Http::STATUS_NOT_FOUND);
+			$response->setStatus(Http::STATUS_NO_CONTENT);
 			return $response;
```
Comment (DeepDiver1975) [inline controller/endpointcontroller.php]: from a pure rest api pov returning a simple empty array would be valid as well - no need for any other error code

Comment (DeepDiver1975) [inline controller/endpointcontroller.php]: ah .. different use case ... hmmm ... ignore my comment

Comment (nickvergessen) [inline controller/endpointcontroller.php]: We use the 204 to stop asking for notifications, when there is no app registered that generates one (default in 8.2)

There is no need to poll every 30 seconds, when nothing can ever potentially show up.

**Comment by davitol** [comment]
:+1: WFM in ownCloud Enterprise Edition 8.2 beta1 (daily) Build:2015-10-05T03:08:46+00:00 2ffb1c2135d80ac82f62c2bd403fe452e2c4e51d

**Pull request merged**